### PR TITLE
added field annotation for read only flag

### PR DESCRIPTION
**Pull request by rolandknight**
Added FieldConfig struct to Annotation to support read only fields. Corresponding change coming to ogent that uses this flag to suppress parameter generation for read only fields (usefult for createdBy and createdAt fields).

### CODE REVIEW

File: entoas/annotation.go [hunk 1]
```diff
@@ -50,6 +52,10 @@ type (
 	}
 	// OperationConfigOption allows managing OperationConfig using functional arguments.
 	OperationConfigOption func(*OperationConfig)
+	// FieldConfig hold meta information aboat a REST field (parameter)
+	FieldConfig struct {
+		ReadOnly bool
+	}
```
Comment (masseelch) [inline entoas/annotation.go:58]: I think we can drop this and add the `ReadOnly` field directly on the `Annotation`.

**Review by masseelch — CHANGES_REQUESTED** [review]
Hey @rolandknight, thank you for this!

Can you please address my comment and add a test for the use case?

File: entoas/annotation.go (hunk 1 above)
Comment (rolandknight) [reply entoas/annotation.go]: Done. 1. ReadOnly directly in Annotation, 2. Builder added, 3. Unit tests added

Once merged, I'll update the corresponding ogent PR

File: entoas/annotation.go [hunk 2]
```diff
@@ -44,6 +42,8 @@ type (
 		Delete OperationConfig
 		// List has meta information about a list operation.
 		List OperationConfig
+		// ReadOnly specifies that the field/edge is read only (no create/update parameter)
```
Comment (masseelch) [inline entoas/annotation.go:45]:
```suggestion
		// ReadOnly specifies that the field/edge is read only (no create/update parameter).
```

File: entoas/annotation.go [hunk 3]
```diff
@@ -104,6 +100,11 @@ func ListOperation(opts ...OperationConfigOption) Annotation {
 	return Annotation{List: operationsConfig(opts)}
 }
 
+// ReadOnly returns a read only field/edge annotation
```
Comment (masseelch) [inline entoas/annotation.go:103]:
```suggestion
// ReadOnly returns a read only annotation.
```

**Review by masseelch — CHANGES_REQUESTED** [review]
Thanks! One thing left to do is to add the `ReadOnly` property to the `Merge()` method.

**Comment by rotemtam** [comment]
Can't we use .Immutable() here ?

**Comment by masseelch** [comment]
> Can't we use .Immutable() here ?

No. Because fields like `created_at` are not immutable. But they should not be part of the generated POST body when attempting to create a new schema object.

**Review by masseelch — APPROVED** [review]
:100: Very well done and thank you very much :blush:

**Pull request merged**